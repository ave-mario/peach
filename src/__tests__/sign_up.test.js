import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SignUp from '../components/sign_up';
import faker from 'faker';

describe('SignUp from:', () => {
  Enzyme.configure({ adapter: new Adapter() });
  const fakePhoneNumber = faker.phone.phoneNumber;
  const fakeName = faker.name.firstName;
  const fakeSurname = faker.name.lastName;
  const props = {
    show: false,
    close: jest.fn(),
    attemtToSignUp: jest.fn(),
  };

  it('component renders correctly', () => {
    const SignUpComponent = renderer.create(<SignUp {...props} />);
    expect(SignUpComponent).toMatchSnapshot();
  });

  it('component renders correctly', () => {
    let newProps = {
      ...props,
      show: true,
    };
    const SignUpComponent = renderer.create(<SignUp {...newProps} />).toJSON();
    expect(SignUpComponent).toMatchSnapshot();
  });

  it('component doesnt render by default', () => {
    const SignUpComponent = mount(<SignUp {...props} />);
    expect(SignUpComponent.prop('show')).toEqual(false);
  });

  it('inputs send correct value to state', () => {
    const SignUpComponent = shallow(<SignUp {...props} />);
    SignUpComponent.find('.phoneNumber').simulate('change', {
      target: { name: 'phoneNumber', value: fakePhoneNumber },
    });
    SignUpComponent.find('.name').simulate('change', {
      target: { name: 'name', value: fakeName },
    });
    SignUpComponent.find('.surname').simulate('change', {
      target: { name: 'surname', value: fakeSurname },
    });
    expect(SignUpComponent.state('phoneNumber')).toEqual(fakePhoneNumber);
    expect(SignUpComponent.state('name')).toEqual(fakeName);
    expect(SignUpComponent.state('surname')).toEqual(fakeSurname);
  });
  it('when form submited - prevent default', () => {
    const SignUpComponent = shallow(<SignUp {...props} />);
    let prevented = false;
    SignUpComponent.find('.formBody').simulate('submit', {
      preventDefault: () => {
        prevented = true;
      },
    });
    expect(prevented).toBe(true);
  });
});
