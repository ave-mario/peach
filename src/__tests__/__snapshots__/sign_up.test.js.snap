// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp from: component renders correctly 1`] = `
<div
  className="sc-htpNat gIsAAs"
>
  <div
    className="sc-bxivhb gPVRgU"
  >
    <h3
      className="sc-ifAKCX bXZmPy"
    >
      SIGN UP
    </h3>
    <span
      className="sc-bwzfXH goCgBH"
      onClick={[MockFunction]}
    >
      ×
    </span>
  </div>
  <form
    className="formBody sc-EHOje fFPbRn"
    onSubmit={[Function]}
  >
    <p>
      Please, enter your phone number:
    </p>
    <input
      className="email sc-bZQynM bGXiwX"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Email"
      value=""
    />
    <input
      className="phoneNumber sc-bZQynM bGXiwX"
      name="phoneNumber"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Phone number"
      value=""
    />
    <input
      className="name sc-bZQynM bGXiwX"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Name"
      value=""
    />
    <input
      className="surname sc-bZQynM bGXiwX"
      name="surname"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Surname"
      value=""
    />
  </form>
  <div
    className="sc-gzVnrw dphOmL"
  >
    <button
      className="sc-bdVaJa fTgRSZ"
      onClick={[MockFunction]}
    >
      CLOSE
    </button>
    <button
      className="sc-bdVaJa iiSudm"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      CONTINUE
    </button>
  </div>
</div>
`;

exports[`SignUp from: component renders correctly 2`] = `
<div
  className="sc-htpNat gIsAAs"
>
  <div
    className="sc-bxivhb gPVRgU"
  >
    <h3
      className="sc-ifAKCX bXZmPy"
    >
      SIGN UP
    </h3>
    <span
      className="sc-bwzfXH goCgBH"
      onClick={[MockFunction]}
    >
      ×
    </span>
  </div>
  <form
    className="formBody sc-EHOje fFPbRn"
    onSubmit={[Function]}
  >
    <p>
      Please, enter your phone number:
    </p>
    <input
      className="email sc-bZQynM bGXiwX"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Email"
      value=""
    />
    <input
      className="phoneNumber sc-bZQynM bGXiwX"
      name="phoneNumber"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Phone number"
      value=""
    />
    <input
      className="name sc-bZQynM bGXiwX"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Name"
      value=""
    />
    <input
      className="surname sc-bZQynM bGXiwX"
      name="surname"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Surname"
      value=""
    />
  </form>
  <div
    className="sc-gzVnrw dphOmL"
  >
    <button
      className="sc-bdVaJa fTgRSZ"
      onClick={[MockFunction]}
    >
      CLOSE
    </button>
    <button
      className="sc-bdVaJa iiSudm"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      CONTINUE
    </button>
  </div>
</div>
`;
